### PR TITLE
op-signer: update client to support Blob tx signing

### PR DESCRIPTION
**Pull request by protolambda**
**Description**

Updates the op-signer client to forward the new EIP-4844 blob attributes. When signing a blob tx with a remote signer, preserve the blobs sidecar locally, but don't forward the full sidecar to the remote.

This also moves the arg-validation into the client package, to avoid the arg-checks not matching the actual args.

Also add some additional logging to the batcher / tx-manager, to confirm we are signing blob-transactions.

**Tests**

Code is covered in the op-signer server-side PR.

### CODE REVIEW

File: op-service/signer/transaction_args.go [hunk 1]
```diff
@@ -62,23 +75,106 @@ func (args *TransactionArgs) data() []byte {
 	return nil
 }
 
-// ToTransaction converts the arguments to a transaction.
-func (args *TransactionArgs) ToTransaction() *types.Transaction {
+func (args *TransactionArgs) Check() error {
```
Comment (trianglesphere) [inline op-service/signer/transaction_args.go:78]: Where do we call this function?

Comment (protolambda) [reply op-service/signer/transaction_args.go:78]: In the server-side (infra repo)

**Review by trianglesphere — APPROVED** [review]

File: op-service/signer/transaction_args.go [hunk 2]
```diff
@@ -28,16 +35,20 @@ type TransactionArgs struct {
 
 	AccessList *types.AccessList `json:"accessList,omitempty"`
 	ChainID    *hexutil.Big      `json:"chainId,omitempty"`
+
+	// Custom extension for EIP-4844 support
+	BlobVersionedHashes []common.Hash `json:"blobVersionedHashes,omitempty"`
+	BlobFeeCap          *hexutil.Big  `json:"blobFeeCap,omitempty"`
```
Comment (protolambda) [inline op-service/signer/transaction_args.go]: Not included in op-geth internal version yet, but in geth v1.13.11 they added it. And called the blob fee cap `maxFeePerBlobGas` in JSON. Will update the PR

**Comment by protolambda** [comment]
Sorry, removed from merge-queue to make a tweak to the JSON field-name to match upstream geth. In op-geth it wasn't included yet, but upstream in latest release it is. Re-queuing the PR now

**Pull request merged**